.github/workflows/ci.yml

```diff
@@ -128,8 +127,23 @@ jobs:
     #     restore-keys: cache-dist-${{ env.CABAL_CACHE_VERSION }}-${{ runner.os }}-${{ matrix.ghc }}
 
     - name: Build dependencies
+      id: build-dependencies
       run: cabal build --only-dependencies all -j
 
+    # Save the `cabal-store` cache if the dependencies were succesfully built,
+    # even if subsequent consensus builds/tests/benchmarks could fail. This
+    # should shave off some running time because dependencies don't have to be
+    # rebuilt if any of the later steps fail and the job has to be re-run.
+    #
+    # See https://github.com/actions/cache/tree/v3/save#always-save-cache.
```

Comment:
INFO: We save built dependencies even if the remainder of the job fails. I hope this will speed up the runs a bit.